### PR TITLE
glib: Don't allow passing a construct(-only) property twice during ob…

### PR DESCRIPTION
**Pull request by sdroege**
…ject construction

GLib also does not allow this and would give a critical warning.

### CODE REVIEW

**Comment by sdroege** [comment]
CC @MathieuDuponchelle

**Comment by sdroege** [comment]
Sure, done

File: glib/src/object.rs [hunk 1]
```diff
@@ -1365,18 +1365,33 @@ impl Object {
...
+                })?;
+
+                if (pspec.flags().contains(crate::ParamFlags::CONSTRUCT)
+                    || pspec.flags().contains(crate::ParamFlags::CONSTRUCT_ONLY))
+                    && properties[0..idx]
+                        .iter()
+                        .any(|(other_name, _)| name == other_name)
+                {
+                    return Err(bool_error!(
+                        "Can't set property '{}' for type '{}' twice",
```
Comment (MathieuDuponchelle) [inline glib/src/object.rs:1382]: Probably worth explaining why ("because it is marked as construct / construct-only")

File: glib/src/object.rs [hunk 2]
```diff
@@ -1395,18 +1410,33 @@ impl Object {
...
```
Comment (MathieuDuponchelle) [inline glib/src/object.rs:1427]: and here :)

File: glib/src/object.rs (hunk 1 above)
Comment (sdroege) [reply glib/src/object.rs]: Good point, done!

**Comment by MathieuDuponchelle** [comment]
patch lgtm

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**